src/dmd/backend/dwarfdbginf.d

```diff
@@ -1438,12 +1453,11 @@ static if (1)
             }
         }
 
-        debugline.total_length = cast(uint)debug_line.buf.length() - 4;
-        memcpy(debug_line.buf.buf, &debugline, debugline.sizeof);
+        rewrite32(debug_line.buf, 0, cast(uint) debug_line.buf.length() - 4);
 
         // Bugzilla 3502, workaround OSX's ld64-77 bug.
         // Don't emit the the debug_line section if nothing has been written to the line table.
-        if (debugline.prologue_length + 10 == debugline.total_length + 4)
+        if (*(cast(uint*) &debug_line.buf[linePrologueLengthOffset]) + 10 == debug_line.buf.length)
```

Comment:
Some machines hit [3502](https://issues.dlang.org/show_bug.cgi?id=3502).
[`/tmp/cirrus-ci-build/linker/ld: internal error in do_relocate_sections, at ../../gold/reloc.cc:955`](https://github.com/bminor/binutils-gdb/blob/91a65d2fe88fabe2d553a0362b2f76034f820175/gold/reloc.cc#L955)